components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -129,6 +130,15 @@ interface Engine {
         onError: ((String, Throwable) -> Unit) = { _, _ -> }
     ): Unit = onError(id, UnsupportedOperationException("Web extension support is not available in this engine"))
 
+    /**
+     * Registers for web extension tab delegation.
```

Comment:
nit: Registers a [WebExtensionTabDelegate] to be notified when web extensions attempt to open/close tabs.